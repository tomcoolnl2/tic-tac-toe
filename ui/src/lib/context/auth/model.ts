import { User } from '@tic-tac-toe/model';
import { AuthError } from './error';

export interface AuthContextState {
	fetchUserName: () => Promise<string | AuthError | null>;
	login: (pwd: string) => Promise<AuthError | User>;
	logout: () => Promise<User>;
}
