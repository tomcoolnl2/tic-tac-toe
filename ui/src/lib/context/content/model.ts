import { Locale } from '@tic-tac-toe/model';

export type ContentAction =
	| { type: 'FETCH_START' }
	| { type: 'FETCH_SUCCESS'; payload: AppContent }
	| { type: 'FETCH_ERROR'; payload: Error }
	| { type: 'SET_LANGUAGE'; payload: Locale };

export interface ContentState {
	appContent: AppContent | null;
	isContentLoading: boolean;
	contentError: Error | null;
	locale: Locale;
}

export type ContentStateWithLanguageSelector = ContentState & {
	setLanguage: (locale: Locale) => void;
};

export interface AppContent {
	appTitle: string;
	loginScreen: AppScreenContent;
	settingsScreen: AppScreenContent;
	gameScreen: AppGameContent;
	restartModal: AppScreenContent;
	gameOverModal: AppScreenContent;
}

export interface AppScreenContent {
	title: string | string[];
	subtitle?: string | string[];
	cta1?: string;
	cta2?: string;
	errors?: {
		[key: string]: string;
	};
}

export interface AppGameContent {
	intelligenceLevel: [string, string, string];
	turnIndicator: [string, string];
	scoreBoard: [string, string, string];
}
