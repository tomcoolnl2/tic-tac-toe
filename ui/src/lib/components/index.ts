export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
