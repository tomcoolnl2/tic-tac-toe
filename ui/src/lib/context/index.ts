export * from './content/content';
export * from './auth/auth';
