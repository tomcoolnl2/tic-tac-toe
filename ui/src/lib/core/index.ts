export * from './canvas';
export * from './grid';
export * from './flex-box';
export * from './divider';
