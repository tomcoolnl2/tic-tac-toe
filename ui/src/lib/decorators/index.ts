export { withTheme } from './withTheme';
export type { ContainerProps } from './withContainer';
export { withContainer } from './withContainer';
