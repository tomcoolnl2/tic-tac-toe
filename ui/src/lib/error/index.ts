export * from './error-boundary';
export { AuthError } from '../context/auth/error';
