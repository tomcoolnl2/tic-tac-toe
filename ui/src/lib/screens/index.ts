export { LoadingScreen } from './loading';
export { LoginScreen } from './login';
export { ErrorScreen } from './error';
export { GameScreen } from './game';
export { SettingsScreen } from './settings';
export { RestartModalScreen } from './reload';
export { GameOverModalScreen } from './game-over';
