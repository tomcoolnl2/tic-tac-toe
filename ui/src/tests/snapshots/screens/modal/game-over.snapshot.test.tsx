import { render } from '@testing-library/react';
import { GameState, PlayerSymbol } from '@tic-tac-toe/model';
import { GameOverModalScreen } from '../../../../lib/screens';

describe('GameOverModalScreen screen component snapshot test', () => {
	const content = {
		title: ['Takes the round', 'No one wins...'],
		subtitle: ["You've won!", "You've lost!", "It's a draw!"],
		cta1: 'Quit',
		cta2: 'Next round',
	};
	it('should match the snapshot', () => {
		const fn = jest.fn();
		const { asFragment } = render(
			<GameOverModalScreen
				content={content}
				gameState={GameState.WIN}
				playerSymbol={PlayerSymbol.X}
				cpuSymbol={PlayerSymbol.O}
				handleQuitGame={fn}
				handleNextRound={fn}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
