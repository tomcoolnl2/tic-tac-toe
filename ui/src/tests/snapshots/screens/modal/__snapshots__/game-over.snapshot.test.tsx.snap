// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverModalScreen screen component snapshot test should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-box direction-column justify-content-flex-start align-items-center game-over"
  >
    <sub>
      You've won!
    </sub>
    <hr
      class="divider full-width horizontal margin-vertical"
    />
    <span
      class="avatar avatar-x avatar-size-l"
    />
    <h2
      class="text-player-x"
    >
      Takes the round
    </h2>
  </div>
  <hr
    class="divider full-width horizontal margin-vertical-l"
  />
  <div
    class="container cols-2 col-gap-l"
    style="--row-height: 1fr; --col-width: 1fr;"
  >
    <button
      class="button button-light"
    >
      Quit
    </button>
    <button
      class="button button-secondary"
    >
      Next round
    </button>
  </div>
</DocumentFragment>
`;
